azurerm/internal/services/loganalytics/log_analytics_clusters_resource.go

```diff
@@ -0,0 +1,397 @@
+	}
+
+	resp, err := client.Get(ctx, resourceGroup, name)
+	if err != nil {
+		return fmt.Errorf("retrieving Log Analytics Cluster %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	if resp.ID == nil || *resp.ID == "" {
+		return fmt.Errorf("empty or nil ID returned for Log Analytics Cluster %q (Resource Group %q) ID", name, resourceGroup)
+	}
+
+	d.SetId(*resp.ID)
+	return resourceArmLogAnalyticsClusterRead(d, meta)
+}
+
+func resourceArmLogAnalyticsClusterRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).LogAnalytics.ClusterClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.LogAnalyticsClusterID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Get(ctx, id.ResourceGroup, id.Name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[INFO] Log Analytics %q does not exist - removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("retrieving Log Analytics Cluster %q (Resource Group %q): %+v", id.Name, id.ResourceGroup, err)
+	}
+	d.Set("name", id.Name)
+	d.Set("resource_group_name", id.ResourceGroup)
+	d.Set("location", location.NormalizeNilable(resp.Location))
+	if err := d.Set("identity", flattenArmLogAnalyticsIdentity(resp.Identity)); err != nil {
+		return fmt.Errorf("setting `identity`: %+v", err)
+	}
+	if props := resp.ClusterProperties; props != nil {
+		if err := d.Set("key_vault_property", flattenArmLogAnalyticsKeyVaultProperties(props.KeyVaultProperties)); err != nil {
+			return fmt.Errorf("setting `key_vault_property`: %+v", err)
+		}
+		d.Set("cluster_id", props.ClusterID)
+	}
+	d.Set("size_gb", flattenArmLogAnalyticsClusterSku(resp.Sku))
+
+	return tags.FlattenAndSet(d, resp.Tags)
+}
+
+func resourceArmLogAnalyticsClusterUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).LogAnalytics.ClusterClient
+	ctx, cancel := timeouts.ForUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.LogAnalyticsClusterID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	parameters := operationalinsights.ClusterPatch{
+		ClusterPatchProperties: &operationalinsights.ClusterPatchProperties{},
+	}
+
+	if d.HasChange("key_vault_property") {
+		parameters.ClusterPatchProperties.KeyVaultProperties = expandArmLogAnalyticsClusterKeyVaultProperties(d.Get("key_vault_property").([]interface{}))
+	}
+
+	if d.HasChange("size_gb") {
+		parameters.Sku = &operationalinsights.ClusterSku{
+			Capacity: utils.Int64(int64(d.Get("size_gb").(int))),
+			Name:     operationalinsights.CapacityReservation,
+		}
+	}
+
+	if d.HasChange("tags") {
+		parameters.Tags = tags.Expand(d.Get("tags").(map[string]interface{}))
+	}
+
+	if _, err := client.Update(ctx, id.ResourceGroup, id.Name, parameters); err != nil {
```

Comment:
Given the note below, can we raise an issue on this as it sounds like it should be an LRO?

Reply:
I can open an issue just to cross our T's and dot our I's, but I have already discussed this with the service team and they have stated that this behavior is currently `By Design` to allow for concurrent operations to be performed on the resource. 🤷‍♂️

Reply:
I think we should, at least that can then be added to the comment below for the official answer and saves us asking the question again at a later date?